treematcher.py

```diff
@@ -307,116 +199,150 @@ def smart_lineage(self, constraint):
 
         return constraint
 
+class TreePattern(Tree):
```

Comment:
TreePattern is now simpler and cleaner, as it only contains the functions to evalue patterns against target nodes/trees. 
Only methods related to that purpose should be added to this class